src/frontend/staff-portal/src/app/components/jj-workbench/jj-dispute-wr-assignments/jj-dispute-wr-assignments.component.ts

```diff
@@ -119,24 +160,33 @@ export class JJDisputeWRAssignmentsComponent implements OnInit, AfterViewInit {
   getAll(team: string): void {
     // filter jj disputes only show new, review, in_progress
     this.data = this.data.filter(x => (this.jjDisputeService.jjDisputeStatusEditable.indexOf(x.status) >= 0) && x.hearingType === this.HearingType.WrittenReasons);
-    this.data = this.data.sort((a, b) => { if (a.submittedTs > b.submittedTs) { return -1; } else { return 1 } });
-    this.resetAssignedUnassigned();
-
-    this.filterByTeam(team); // initialize
-  }
+    this.data = this.data.sort((a, b) => {
+      if (((a.jjAssignedTo === this.valueOfUnassigned || !a.jjAssignedTo) && (b.jjAssignedTo === this.valueOfUnassigned || !b.jjAssignedTo))
+     || (a.jjAssignedTo !== this.valueOfUnassigned && b.jjAssignedTo !== this.valueOfUnassigned)) {
 
-  resetAssignedUnassigned() {
-    this.assignedDataSource.data = null;
-    this.unassignedDataSource.data = null;
+      if (a.submittedTs <= b.submittedTs) return -1;
+      else return 1;
+    }
+      let firstKeyResult = 1;
+      if (a.jjAssignedTo == this.valueOfUnassigned || !a.jjAssignedTo) firstKeyResult = -1;
+      return firstKeyResult;
+    });
 
-    this.assignedDataSource.data = this.data.filter(x => x.jjAssignedTo && x.jjAssignedTo !== this.valueOfUnassigned);
-    this.unassignedDataSource.data = this.data.filter(x => !x.jjAssignedTo || x.jjAssignedTo === this.valueOfUnassigned);
-    this.unassignedDataSource.data.forEach(jjDispute => {
-      let index = this.unassignedDataSource.data.indexOf(jjDispute);
-      jjDispute.jjAssignedTo = this.valueOfUnassigned;
-      this.unassignedDataSource.data[index] = jjDispute;
+    this.dataSource.data = this.data;
+    this.dataSource.data.forEach(jjDispute => {
+      if (!jjDispute.jjAssignedTo) {
+        let index = this.dataSource.data.indexOf(jjDispute);
+        jjDispute.jjAssignedTo = this.valueOfUnassigned;
+        this.dataSource.data[index] = jjDispute;
+      }
     });
+    this.dataSource.data = [... this.dataSource.data]; // trigger sort
```

Comment:
If this trigger the sort, line 167 - 173 can be removed?
and should use `this.dataSource.sort = this.sort;` to trigger sorting

Reply:
done